cmd/gofr/migration/dbMigration/clickhouse.go

```diff
@@ -0,0 +1,179 @@
+package dbmigration
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"gofr.dev/pkg/datastore"
+	"gofr.dev/pkg/errors"
+	"gofr.dev/pkg/log"
+)
+
+type Clickhouse struct {
+	datastore.ClickHouseDB
+	existingMigration []gofrMigration
+}
+
+func NewClickhouse(c datastore.ClickHouseDB) *Clickhouse {
+	return &Clickhouse{c, make([]gofrMigration, 0)}
+}
+
+// Run executes a migration
+func (c *Clickhouse) Run(m Migrator, app, name, methods string, logger log.Logger) error {
+	if c.Conn == nil {
+		return errors.DataStoreNotInitialized{DBName: datastore.ClickHouse}
+	}
+
+	err := c.preRun(app, methods, name)
+	if err != nil {
+		return err
+	}
+
+	ds := &datastore.DataStore{ClickHouse: c.ClickHouseDB}
+
+	if methods == UP {
+		err = m.Up(ds, logger)
+	} else {
+		err = m.Down(ds, logger)
+	}
+
+	if err != nil {
+		return &errors.Response{Reason: "error encountered in running the migration", Detail: err}
+	}
+
+	err = c.postRun(app, methods, name)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Clickhouse) preRun(app, method, name string) error {
+	const migrationTableSchema = `CREATE TABLE IF NOT EXISTS gofr_migrations (
+    app String,
+    version Int64,
```

Comment:
can we use an Unsigned integer? Since we are storing unix timestamps as versions